### PR TITLE
EVNT-43: Allow to use external ActiveMQ insatnce

### PR DESCRIPTION
**Pull request by afijal**
EVNT-43: Allow to use external ActiveMQ insatnce
This PR allows to use external ActiveMQ insatnce by setting **activeMQ.externalUrl** global property, if not set or empty then fallback to default behavior (embedded ActiveMQ)

issue link:
https://issues.openmrs.org/browse/EVNT-43

### CODE REVIEW

**Comment by HerbertYiga** [comment]
well-done on this @afijal ,you can also add a ticket Id on this

**Comment by afijal** [comment]
Thanks @HerbertYiga, just added it

**Comment by dkayiwa** [comment]
Can you also add a link to the pull request as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

Secondly, was this a problem that you face with an implementation site?

**Comment by afijal** [comment]
@dkayiwa Sure, added.
Yes, we wanted to use external ActiveMQ in our implementation but found no solutions and thought that it might be useful for someone else as well.

File: api/src/main/java/org/openmrs/event/EventEngine.java [hunk 1]
```diff
@@ -121,25 +123,32 @@ public Message createMessage(Session session) throws JMSException {
...
+                try {
+                    dataDirectory = URLEncoder.encode(dataDirectory, "UTF-8");
+                }
+                catch (UnsupportedEncodingException e) {
+                    throw new RuntimeException("Failed to encode URI", e);
+                }
+                brokerURL = "vm://localhost?broker.persistent=true&broker.useJmx=false&broker.dataDirectory="
+                    + dataDirectory;
+            } else {
+                brokerURL = "tcp://" + property;
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/event/EventEngine.java:143]: Don't you want the property to have the full url set in the global property?

Comment (afijal) [reply api/src/main/java/org/openmrs/event/EventEngine.java:143]: I didn't want to require end users to be that technical, and I guess tcp is the most commonly used one

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/event/EventEngine.java:143]: Fair enough. 😊

**Pull request merged**